### PR TITLE
[Owner Review] Added Session Tests for NI-DMM

### PR DESCRIPTION
**Pull request by ShantanuShinde**
### What does this Pull Request accomplish?

The Session test tests 3 DMM APIs: `InitWithOptions`, `Close` and `GetErrorMessage`
This has 6 tests:
1. Initiate a Session with device session name
2. Initiate session with device but without session name
3. Initiate session without device, returning appropriate error code
4. Closing a valid session
5. Closing invalid session
6. Initialize without device and check if error message is correct

### Why should this Pull Request be merged?

These tests helps ensures the DMM APIs are working properly

### What testing has been done?

The tests have been run locally

TODO: Detail what testing has been done to ensure this submission meets requirements.

### CODE REVIEW

**Review by kruthi-NI — APPROVED** [review]

**Review by ganeshnie — APPROVED** [review]

**Review by Akshata440 — APPROVED** [review]

**Review by reckenro — APPROVED** [review]

File: CMakeLists.txt
Comment (SparkingSpork) [inline CMakeLists.txt]: We should probably keep the driver test files here in alphabetical order. E.g.:
```cmake
    "source/tests/system/session_utilities_service_tests.cpp"
    "source/tests/system/nidmm_session_tests.cpp"
    "source/tests/system/niscope_driver_api_tests.cpp"
    "source/tests/system/niscope_session_tests.cpp"
    "source/tests/system/niswitch_driver_api_tests.cpp"
    "source/tests/system/niswitch_session_tests.cpp"
    "source/tests/system/nisync_session_tests.cpp"
```

**Review by SparkingSpork — APPROVED** [review]

**Review by jonlugoatx — CHANGES_REQUESTED** [review]
These changes look good, but do not merge them into `users/sshinde/dmm_autogen`. Wait until that PR gets approved and submitted, then you can merge this into the feature branch. Once that PR is in, then you can reset my status and submit.

File: CMakeLists.txt
Comment (ShantanuShinde) [reply CMakeLists.txt]: ordered it alphabetically

**Review by jonlugoatx — APPROVED** [review]

**Pull request merged**
